crates/crypto/proof-params/src/lib.rs

```diff
@@ -17,12 +19,52 @@ pub use traits::{
 #[cfg(feature = "proving-keys")]
 mod proving_keys;
 
-/// Proving key for the spend proof.
 #[cfg(feature = "proving-keys")]
-pub static SPEND_PROOF_PROVING_KEY: Lazy<ProvingKey<Bls12_377>> =
-    Lazy::new(proving_keys::spend_proving_parameters);
+#[derive(Debug, Default)]
+pub struct LazyProvingKey {
+    inner: OnceCell<ProvingKey<Bls12_377>>,
+}
+
+#[cfg(feature = "proving-keys")]
+impl LazyProvingKey {
+    // Lazily construct proving key
+    pub fn new() -> Self {
+        LazyProvingKey { 
+            inner: OnceCell::new(),
+        }
+    }
+
+    pub fn load(&self, key: &str) -> ProvingKey<Bls12_377> {
+        self.inner.get_or_init(|| match key {
+            "spend_key" => proving_keys::spend_proving_parameters(),
+            _ => todo!()
+        }).clone()
+    }
+}
+
+#[cfg(feature = "proving-keys")]
+impl Deref for LazyProvingKey {
+    type Target = ProvingKey<Bls12_377>;
+
+    fn deref(&self) -> &Self::Target {
+        self.inner.get().expect("Proving key cannot be loaded!")
+    }
+}
 
-/// Verifying key for the spend proof.
+/// Note: Conditionally load the proving key objects in the crate
+/// unless the target is wasm. 
+
+/// Proving key for the spend proof.
+pub static SPEND_PROOF_PROVING_KEY: Lazy<LazyProvingKey> = Lazy::new(|| {
+    let mut spend_proving_key = LazyProvingKey::new();
+
+    #[cfg(all(feature = "proving-keys", not(target_arch = "wasm32")))]
+    spend_proving_key.inner.get_or_init(proving_keys::spend_proving_parameters);
```

Comment:
Here we can do this or we can call `load` with the bytes from e.g. `include_bytes!("gen/spend_pk.bin")`